### ISSUE TITLE
[Bug] container unable to start without providing ubound.conf

### ISSUE THREAD

**Issue by cacula** [madnuttah/unbound-docker#51]
**Describe the bug**
After the 1.19.1-0 update the container does not start anymore and gets into a crash loop. following error appears in the logs:
```
[1707896685] unbound[1:0] error: can't bind socket: Address not available for ::1 port 53
[1707896685] unbound[1:0] fatal error: could not open ports
```
**To Reproduce**
Use latest version with following compose file
```
  unbound:
    container_name: unbound
    hostname: unbound
    image: madnuttah/unbound:latest
    restart: unless-stopped
    deploy:
      resources:
        limits:
          memory: 50m
    ports:
      - "127.0.0.1:5335:5335/tcp"
      - "127.0.0.1:5335:5335/udp"
    networks:
      - dns
    environment:
      TZ: Europe/Berlin
```
**Expected behavior**
container should start as it did before the update

**Workaround**
Download "default" unbound.conf from [https://github.com/madnuttah/unbound-docker/blob/main/unbound/root/usr/local/unbound/unbound.conf ](https://github.com/madnuttah/unbound-docker/blob/main/unbound/root/usr/local/unbound/unbound.conf ) and add it to the compose file:
```
  unbound:
    container_name: unbound
    hostname: unbound
    image: madnuttah/unbound:latest
    restart: unless-stopped
    deploy:
      resources:
        limits:
          memory: 50m
    ports:
      - "127.0.0.1:5335:5335/tcp"
      - "127.0.0.1:5335:5335/udp"
    networks:
      - dns
    environment:
      TZ: Europe/Berlin
    volumes:
      - ./unbound.conf:/usr/local/unbound/unbound.conf:rw
```
**Please complete the following information:**
 - Device: VM
 - OS: Debian Bookworm
 - Architecture: amd64
 - Version: latest

**Comment by madnuttah** [comment]
Hi and thanks for opening a new issue. Take a look at your config files. Do a quick search for "53" and comment out those lines. The image is way more secure now because unbound runs in a pretty limited user context. You would need to set capacities in your docker compose if you insist in using port 53.

Would you mind showing your configs without sensitive information? I'll take a look then, there are maybe more entries to adapt. Thank you.

**Comment by cacula** [comment]
that's my complete docker compose
```
# https://github.com/pi-hole/docker-pi-hole
# https://github.com/madnuttah/unbound-docker
version: "3"

services:
  pihole:
    container_name: pihole
    hostname: pihole
    image: jacklul/pihole:latest
    restart: unless-stopped
    depends_on:
      - unbound
    deploy:
      resources:
        limits:
          memory: 250m
    ports:
      - "192.168.1.254:53:53/tcp"
      - "192.168.1.254:53:53/udp"
    networks:
      - traefik
      - dns
    environment:
      TZ: Europe/Berlin
      FTLCONF_LOCAL_IPV4: 192.168.1.254
      PIHOLE_DNS_: unbound#5335
      REV_SERVER: true
      REV_SERVER_DOMAIN: xxx
      REV_SERVER_TARGET: 192.168.1.1
      REV_SERVER_CIDR: 10.8.0.0/16
      WEBTHEME: default-auto
      DNSMASQ_LISTENING: all
    volumes:
       - etc:/etc/pihole/
       - dnsmasq_d:/etc/dnsmasq.d/
       - updatelists:/etc/pihole-updatelists/
       - ./pihole-updatelists.conf:/etc/pihole-updatelists/pihole-updatelists.conf
    labels:
       - traefik.enable=true
       - traefik.http.routers.pihole-ssl.rule=Host(`pihole.example.net`)
       - traefik.http.routers.pihole-ssl.entrypoints=websecure
       - traefik.http.routers.pihole-ssl.tls=true
       - traefik.http.routers.pihole-ssl.tls.certresolver=myresolver
       - traefik.http.routers.pihole-ssl.service=pihole-ui
       - traefik.http.routers.pihole.rule=Host(`pihole.example.net`)
       - traefik.http.routers.pihole.entrypoints=web
       - traefik.http.routers.pihole.service=pihole-ui
       - traefik.http.services.pihole-ui.loadbalancer.server.port=80
  unbound:
    container_name: unbound
    hostname: unbound
    image: madnuttah/unbound:latest
    restart: unless-stopped
    deploy:
      resources:
        limits:
          memory: 50m
    ports:
      - "127.0.0.1:5335:5335/tcp"
      - "127.0.0.1:5335:5335/udp"
    networks:
      - dns
    environment:
      TZ: Europe/Berlin
    volumes:
      - ./unbound.conf:/usr/local/unbound/unbound.conf:rw

volumes:
  etc:
  dnsmasq_d:
  updatelists:

networks:
  traefik:
    external: true
  dns:
```
as mentioned in the bug report, the following line I only had to add  as a workaround to get the container running again
```
    volumes:
      - ./unbound.conf:/usr/local/unbound/unbound.conf:rw
```
the unbound.conf I'm using for the workaround is just a fresh copy I downloaded from your repo - without any modifications: [https://github.com/madnuttah/unbound-docker/blob/main/unbound/root/usr/local/unbound/unbound.conf](https://github.com/madnuttah/unbound-docker/blob/main/unbound/root/usr/local/unbound/unbound.conf)
I just want to use Port 53 for pihole as it will serve as DNS on the local network.

**Comment by madnuttah** [comment]
I'm sorry I wasn't clear enough, I meant your unbound.conf(s) but I see an issue here. You try to access [127.0.0.1](localhost). This won't work anymore because of the limited user account without special permissions. Can you set the unbound ip address instead `127.0.0.1`?

Please drop your configs if you don't mind. 

Thank you

**Comment by cacula** [comment]
I decided to keep using my own unbound.conf from now on which is probably a good idea anyway. that's my conf and works as expected
```
server:
	num-threads: 1
	interface: 0.0.0.0@5335
	prefer-ip6: no
	prefer-ip4: yes
	outgoing-range: 4096
	outgoing-port-permit: 32768
	msg-cache-size: 4m
	msg-cache-slabs: 4
	num-queries-per-thread: 4096
	udp-connect: yes
	rrset-cache-size: 4m
	rrset-cache-slabs: 4
	cache-min-ttl: 0
	cache-max-ttl: 86400
	infra-cache-slabs: 4
	do-ip4: yes
	do-ip6: no
	do-udp: yes
	do-tcp: yes
	use-systemd: no
	do-daemonize: no
	access-control: 0.0.0.0/0 allow
	chroot: ""
	username: ""
	directory: "/usr/local/unbound/"
	logfile: ""
	use-syslog: no
	log-time-ascii: yes
	log-servfail: yes
	pidfile: "/usr/local/unbound/unbound.d/unbound.pid"
	hide-identity: yes
	hide-version: yes
	hide-trustanchor: yes
	harden-short-bufsize: yes
	harden-large-queries: yes
	harden-glue: yes
	harden-dnssec-stripped: yes
	harden-below-nxdomain: yes
	harden-algo-downgrade: yes
	qname-minimisation: yes
	aggressive-nsec: yes
	use-caps-for-id: no
	private-address: 10.0.0.0/8
	private-address: 172.16.0.0/12
	private-address: 192.168.0.0/16
	private-address: 169.254.0.0/16
	private-address: fd00::/8
	private-address: fe80::/10
	private-address: ::ffff:0:0/96
	do-not-query-localhost: yes
	deny-any: yes
	rrset-roundrobin: yes
	minimal-responses: yes
	disable-dnssec-lame-check: no
	module-config: "validator iterator"
	auto-trust-anchor-file: "/usr/local/unbound/iana.d/root.key"
	trust-anchor-signaling: yes
	root-key-sentinel: yes
	val-max-restart: 5
	val-clean-additional: yes
	val-permissive-mode: no
	ignore-cd-flag: no
	serve-expired: no

	# Have the validator log failed validations for your diagnosis.
	# 0: off. 1: A line per failed user query. 2: With reason and bad IP.
	val-log-level: 0

	zonemd-permissive-mode: no
	key-cache-size: 4m
	key-cache-slabs: 4
	neg-cache-size: 1m
	unblock-lan-zones: no
	insecure-lan-zones: yes

auth-zone:
	name: "."
	primary: 170.247.170.2        # b.root-servers.net
	primary: 192.33.4.12          # c.root-servers.net
	primary: 199.7.91.13          # d.root-servers.net
	primary: 192.5.5.241          # f.root-servers.net
	primary: 192.112.36.4         # g.root-servers.net
	primary: 193.0.14.129         # k.root-servers.net
	primary: 192.0.47.132         # xfr.cjr.dns.icann.org
	primary: 192.0.32.132         # xfr.lax.dns.icann.org
	primary: 2801:1b8:10::b       # b.root-servers.net
	primary: 2001:500:2::c        # c.root-servers.net
	primary: 2001:500:2d::d       # d.root-servers.net
	primary: 2001:500:2f::f       # f.root-servers.net
	primary: 2001:500:12::d0d     # g.root-servers.net
	primary: 2001:7fd::1          # k.root-servers.net
	primary: 2620:0:2830:202::132 # xfr.cjr.dns.icann.org
	primary: 2620:0:2d0:202::132  # xfr.lax.dns.icann.org
	#url: "https://www.internic.net/domain/root.zone"
	fallback-enabled: yes
	for-downstream: no
	for-upstream: yes
	zonemd-check: yes
    zonemd-reject-absence: no
	zonefile: "/iana.d/root.zone"
```
thanks for your project & support - very appreciated!

**Comment by madnuttah** [comment]
Yes, you found a bug. Will be fixed soon, many thanks!

**Comment by cacula** [comment]
@madnuttah I can confirm that 1.19.1-2 fixed it - thanks for the fast reaction!

**Comment by madnuttah** [comment]
Thank you for reporting it! Glad it runs for you.
